src/main/java/com/shootdoori/match/entity/TeamMember.java

```diff
@@ -94,4 +98,27 @@ public LocalDateTime getUpdatedAt() {
     public void setTeam(Team team) {
         this.team = team;
     }
+
+    public void changeRole(Team team, TeamMemberRole newRole) {
+
+        if (newRole == TeamMemberRole.LEADER && team.hasCaptain()
+            && this.role != TeamMemberRole.LEADER) {
+            throw new DuplicateCaptainException();
+        }
+
+        if (newRole == TeamMemberRole.VICE_LEADER && team.hasViceCaptain()
+            && this.role != TeamMemberRole.VICE_LEADER) {
+            throw new DuplicateViceCaptainException();
+        }
+
+        this.role = newRole;
+    }
```

Comment:
```suggestion
    public void changeRole(Team team, TeamMemberRole newRole) {
        if (isPromotionToLeader(newRole)) {
            validateLeaderPromotion(team);
        }
    
        if (isPromotionToViceLeader(newRole)) {
            validateViceLeaderPromotion(team);
        }
            this.role = newRole;
    }
```

```java
private boolean isPromotionToLeader(TeamMemberRole newRole) {
    return newRole == TeamMemberRole.LEADER && this.role != TeamMemberRole.LEADER;
}

private boolean isPromotionToViceLeader(TeamMemberRole newRole) {
    return newRole == TeamMemberRole.VICE_LEADER && this.role != TeamMemberRole.VICE_LEADER;
}

private void validateLeaderPromotion(Team team) {
    if (team.hasCaptain()) {
        throw new DuplicateCaptainException("팀에 이미 리더가 존재합니다.");
    }
}

private void validateViceLeaderPromotion(Team team) {
    if (team.hasViceCaptain()) {
        throw new DuplicateViceCaptainException("팀에 이미 부리더가 존재합니다.");
    }
}
```

- 아이코.. 전 머리가 나빠서 정책이 좀만 겹쳐도 메서드 추출하는걸 선호합니다.